tsdate/rescaling.py

```diff
@@ -110,39 +112,257 @@ def f(i, j):  # loss
+#
+#    indexes_mutation = np.argsort(mutations_position)
+#    position_insert = edges_left[indexes_insert]
+#    position_remove = edges_right[indexes_remove]
+#    position_mutation = mutations_position[indexes_mutation]
+#
+#    nodes_edge = np.full(num_nodes, tskit.NULL)
+#    mutations_edge = np.full(num_mutations, tskit.NULL)
+#    edges_mutations = np.zeros(num_edges)
+#    edges_span = edges_right - edges_left
+#
+#    left = 0.0
+#    a, b, d = 0, 0, 0
+#    while a < num_edges or b < num_edges:
+#        while b < num_edges and position_remove[b] == left:  # edges out
+#            e = indexes_remove[b]
+#            c = edges_child[e]
+#            nodes_edge[c] = tskit.NULL
+#            b += 1
+#
+#        while a < num_edges and position_insert[a] == left:  # edges in
+#            e = indexes_insert[a]
+#            c = edges_child[e]
+#            nodes_edge[c] = e
+#            a += 1
+#
+#        right = sequence_length
+#        if b < num_edges:
+#            right = min(right, position_remove[b])
+#        if a < num_edges:
+#            right = min(right, position_insert[a])
+#        left = right
+#
+#        while d < num_mutations and position_mutation[d] < right:
+#            m = indexes_mutation[d]
+#            c = mutations_node[m]
+#            e = nodes_edge[c]
+#            if e != tskit.NULL:
+#                mutations_edge[m] = e
+#                edges_mutations[e] += 1.0
+#            d += 1
+#
+#    mutations_edge = mutations_edge.astype(np.int32)
+#    edges_stats = np.column_stack((edges_mutations, edges_span))
+#
+#    return edges_stats, mutations_edge
+#
+#
+# def count_mutations(ts, constraints=None):
+#    """
+#    Return an array with `num_edges` rows, and columns that are the number of
+#    mutations per edge and the total span per edge
+#    """
+#    # TODO: adjust spans by an accessibility mask
+#    if constraints is None:
+#        node_is_leaf = np.full(ts.num_nodes, False)
+#        node_is_leaf[list(ts.samples())] = True
+#    else:
+#        assert constraints.shape == (ts.num_nodes, 2)
+#        node_is_leaf = np.logical_and(
+#            constraints[:, 0] == 0.0,
+#            constraints[:, 0] == constraints[:, 1],
+#        )
+#    return _count_mutations(
+#        node_is_leaf,
+#        ts.mutations_node,
+#        ts.sites_position[ts.mutations_site],
+#        ts.edges_parent,
+#        ts.edges_child,
+#        ts.edges_left,
+#        ts.edges_right,
+#        ts.indexes_edge_insertion_order,
+#        ts.indexes_edge_removal_order,
+#        ts.sequence_length,
+#    )
+
+
+@numba.njit(
+    _tuple((_f2w, _i1w))(_b1r, _i1r, _f1r, _i1r, _i1r, _f1r, _f1r, _i1r, _i1r, _f, _b)
+)
+def _count_mutations(
+    node_is_sample,
+    mutations_node,
+    mutations_position,
+    edges_parent,
+    edges_child,
+    edges_left,
+    edges_right,
+    indexes_insert,
+    indexes_remove,
+    sequence_length,
+    size_biased,
+):
+    """
+    Internals for `count_mutations`
```

Comment:
Could probably have this as a comment (# xxx) rather than a docstring, as it's not intended to be exposed in documentation?